### PR TITLE
adding searching feature

### PR DESCRIPTION
**Pull request by annababilon**

### CODE REVIEW

File: src/components/CreateTodoModal.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ export default function CreateTodoModal({
   addTodo,
 }) {
   const newTodo = useRef();
+  const newTodoDescription =useRef();
```
Comment (JMB98) [inline src/components/CreateTodoModal.js:15]: new reference `newTodoDescription` which is never used.

File: src/components/CreateTodoModal.js [hunk 2]
```diff
@@ -56,7 +57,10 @@ export default function CreateTodoModal({
 
         <FontAwesomeIcon icon = {faWindowClose} size="3x" className="close-todo-btn" onClick={closeCreateTodo}></FontAwesomeIcon>
         </div>
+        <h4>Title:</h4>
         <input className="todo-input" type="text" ref={newTodo}></input>
+        {/* <h4>Description:</h4>
+        <input className="todo-input" type="text" ref={newTodoDescription}></input> */}
```
Comment (JMB98) [inline src/components/CreateTodoModal.js]: delete it if not needed anymore

File: src/components/TodoPanel.js [hunk 3]
```diff
@@ -60,6 +61,18 @@ export default function TodoPanel({setTodosAmount}) {
...
+        const searchedPhrase = phrase.current.value;
+        setToSearch(searchedPhrase);
+    }
+
+    function cleanInput() {
+        setToSearch("");
+        phrase.current.value = "";
+     
+
+    }
```
Comment (JMB98) [inline src/components/TodoPanel.js]: unecessary new lines

File: src/components/TodoPanel.js [hunk 4]
```diff
@@ -75,12 +88,12 @@ export default function TodoPanel({setTodosAmount}) {
             <CreateTodoModal isOpen={isTodoModalOpen} closeCreateTodo = {()=> setIsTodoModalOpen(false)} addTodo={addTodo}/>
             <div>
                 <div className = 'search-todo-panel'>
-                    <input className = 'search-todo-input' placeholder='enter a phrase that you are searching for'></input>
-                    {/* <i className="fas fa-search fa-1g"></i> */}
-                    <FontAwesomeIcon icon = {faSearch} className='fa-search'/>
+                    <input className = 'search-todo-input' placeholder='enter a phrase that you are searching for'  ref = {phrase} onChange = {searchTodos}></input>
+                    
+                    {toSearch? <FontAwesomeIcon icon = {faTimes}  className='fa-times' onClick = {cleanInput}/> : <FontAwesomeIcon icon = {faSearch} className='fa-search'/> }
```
Comment (JMB98) [inline src/components/TodoPanel.js]: search icon has no functionality. I see two possibilities:
1. display no icon as long as there is no input
2. display the search icon and when the user clicks on it, while there is empty input, display a tooltip which says something like "please insert a query".

File: src/components/TodoPanel.js [hunk 5]
```diff
@@ -75,12 +88,12 @@ export default function TodoPanel({setTodosAmount}) {
             <CreateTodoModal isOpen={isTodoModalOpen} closeCreateTodo = {()=> setIsTodoModalOpen(false)} addTodo={addTodo}/>
             <div>
                 <div className = 'search-todo-panel'>
-                    <input className = 'search-todo-input' placeholder='enter a phrase that you are searching for'></input>
-                    {/* <i className="fas fa-search fa-1g"></i> */}
-                    <FontAwesomeIcon icon = {faSearch} className='fa-search'/>
+                    <input className = 'search-todo-input' placeholder='enter a phrase that you are searching for'  ref = {phrase} onChange = {searchTodos}></input>
```
Comment (JMB98) [inline src/components/TodoPanel.js]: "enter" => "Enter"

File: src/styles.css [hunk 6]
```diff
@@ -285,9 +290,19 @@ font-weight: 500;
...
+  padding: 5px;
+  margin: 5px 10px;
+  font-size: 15px ;
+  font-weight: 500;
+  color: #3c403d;
+  border: solid 2px #8eada7;
+  border-radius: 3px;
+  cursor: pointer;
+} */
+
```
Comment (JMB98) [inline src/styles.css]: will you need it again? If not, delete it

**Comment by JMB98** [comment]
everything looks nice and searching feature works fine :ok_hand: . No bigger issues.

File: src/components/CreateTodoModal.js (hunk 1 above)
Comment (annababilon) [reply src/components/CreateTodoModal.js:15]: I did it - see in the next merge request

File: src/components/TodoPanel.js (hunk 4 above)
Comment (annababilon) [reply src/components/TodoPanel.js]: it will stay as we discussed

**Pull request merged**